server/server.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"log"
+	"net"
+	"net/rpc"
+	"sync"
+
+	"github.com/lukedmor/conduit_tiny/common"
+)
+
+type Provider struct {
+	cs *conduitServer
+}
+
+func (p *Provider) Join(a *common.ProviderJoinLeaveArgs, reply *common.Nothing) error {
+	p.cs.providers[a.Addr] = true
+	return nil
+}
+
+func (p *Provider) Leave(a *common.ProviderJoinLeaveArgs, reply *common.Nothing) error {
+	delete(p.cs.providers, a.Addr)
```

Comment:
I agree that the client should ping the provider. I don't know that the server should verify this. That might put some strain on the server if it has a hand in confirmation of aliveness.